locator/network_topology_strategy.cc

```diff
@@ -376,7 +377,15 @@ future<tablet_replica_set> network_topology_strategy::reallocate_tablets(schema_
         ++nodes_per_dc[node.dc_rack().dc];
     }
 
-    for (const auto& [dc, dc_rf] : _dc_rep_factor) {
+    // #22688 - take all dcs in topology into account when detemining migration.
```

Comment:
nit: pleass fix spelling.

And, I don't understand the motivation.
Why not simply continue the loop if dc_rf==0?

Reply:
I think a more elegant approach would be to change the `nodes_per_dc` map to track both the node count and the rf and so it will contain all the information we need for the loop below, without having to call get_replication_factor for each dc.

Reply:
Can extend `nodes_per_dc` with zeroes to have it to iterate over instead of building a set here, but will still need to call `get_replication_factor` for each, since we need to get rf or none=0 from the `_dc_rep_factor` (this is what `get_replication_factor` does).

Reply:
Otoh, could just iterate the full topology and let the rf/node count check do the work.

Reply:
we need to search the dc either in nodes_per_dc or in _dc_rep_factor

Reply:
my thinking was to populate nodes_per_dc as we do today and then, traversing dc_rep_factor, patch-in the rf per dc.
We can also lookup the rffo4 each dc in the first loop and emplace the entry once per dc